SpotifyAPI/Local/SpotifyLocalAPI.cs

```diff
@@ -149,21 +157,72 @@ public StatusResponse GetStatus()
         }
 
         /// <summary>
-        /// Mutes Spotify (Requires nircmd.dll)
+        /// Mutes Spotify (Requires Windows 7 or newer)
         /// </summary>
         public void Mute()
         {
-            if (File.Exists("nircmd.dll"))
-                DoNirCmd("muteappvolume spotify.exe 1");
+            //Windows < Windows Vista Check
+            Contract.Requires(Environment.OSVersion.Version.Major >= 6);
+            //Windows Vista Check
+            if (Environment.OSVersion.Version.Major == 6)
+                Contract.Requires(Environment.OSVersion.Version.Minor != 0);
+            VolumeMixerControl.MuteSpotify(true);
         }
 
         /// <summary>
-        /// Unmutes Spotify (Requires nircmd.dll)
+        /// Unmutes Spotify (Requires Windows 7 or newer)
         /// </summary>
         public void UnMute()
         {
-            if (File.Exists("nircmd.dll"))
-                DoNirCmd("muteappvolume spotify.exe 0");
+            //Windows < Windows Vista Check
+            Contract.Requires(Environment.OSVersion.Version.Major >= 6);
+            //Windows Vista Check
+            if (Environment.OSVersion.Version.Major == 6)
+                Contract.Requires(Environment.OSVersion.Version.Minor != 0);
+            VolumeMixerControl.MuteSpotify(false);
+        }
+
+        /// <summary>
+        /// Checks whether Spotify is muted in the Volume Mixer control (required Windows 7 or newer)
+        /// </summary>
+        /// <returns>Null if an error occured, otherwise the muted state</returns>
+        public bool? IsSpotifyMuted()
```

Comment:
This is also a possibility, this way however you would not have to catch an Exception but simply check the return value. I just read into Design Guidelines for using Exceptions, they also advise to throw Exceptions instead of returning something similar to a HRESULT. I will change this